opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1362,73 +1362,117 @@ template cpgrid::Entity<3> Dune::createEntity(const CpGrid&, int, bool);
-                 parent_to_children_cells, child_to_parent_faces, child_to_parent_cells]
-        = (*(this-> data_[0])).refinePatch(cells_per_dim, startIJK, endIJK);
-    // Add level 1 to "data".
-    (this-> data_).push_back(level1_ptr);
-    //
-    // LEVEL 0, definition/declaration of some members:
-    (*data_[0]).level_data_ptr_ = &(this -> data_);
-    (*data_[0]).level_ = 0;
-    // Relation between level and leafview cell indices.
-    std::map<int,int>& l0_to_leaf_cells = (*data_[0]).level_to_leaf_cells_;
-    // For level0, attach children to each parent cell. EMPTY entry for no parents.
+    // Total amount of patches:
+    const int& num_patches = startIJK_vec.size();
+    assert(cells_per_dim_vec.size() == startIJK_vec.size());
+    assert(cells_per_dim_vec.size() == endIJK_vec.size());
+    std::vector<int> all_patch_corners; 
+    std::vector<int> all_patch_faces; 
+    std::vector<int> all_patch_cells;
+    std::vector<std::vector<std::array<int,2>>> all_patch_boundary_old_to_new_corners;
+    std::vector<std::vector<std::tuple<int,std::vector<int>>>> all_patch_boundary_old_to_new_faces;
+    std::vector<std::vector<std::tuple<int,std::vector<int>>>> all_patch_parent_to_children_cells;
+    std::vector<std::vector<std::array<int,2>>> all_patch_child_to_parent_cells;
+    all_patch_boundary_old_to_new_corners.reserve(num_patches);
+    all_patch_boundary_old_to_new_faces.reserve(num_patches);
+    all_patch_parent_to_children_cells.resize(num_patches);
+    all_patch_child_to_parent_cells.resize(num_patches);
+    // For level0, attach children to each parent cell. For no parents, entry {-1, {-1}} representing {no level, {no children}}
     auto& l0_parent_to_children_cells = (*data_[0]).parent_to_children_cells_;
-    const std::vector<int>& no_child = {-1};
-    const int& no_level = -1;
-    const std::tuple<int, std::vector<int>>& no_children = std::make_tuple(no_level, no_child);
-    // For cells with no children, we set {-1,{-1}}. Entries of actual parents will be rewritten.
     l0_parent_to_children_cells.resize(data_[0]-> size(0));
-    for (int cell = 0; cell < data_[0] -> size(0); ++cell){
-        l0_parent_to_children_cells[cell] = no_children;
-    }
-    // For level1/LGR, attach to each child-cell its parent.
-    std::vector<std::array<int,2>>& l1_child_to_parent_cells = (*data_[1]).child_to_parent_cells_;
-    // For cells with no parent, we set {-1,-1}. Entries with actual parents will be rewritten.
-    const std::array<int,2>& no_parent = {-1,-1};
-    l1_child_to_parent_cells.resize(data_[1]-> size(0));
-    for (int cell = 0; cell < data_[1]->size(0); ++cell){
-        l1_child_to_parent_cells[cell] = no_parent;
+    const std::vector<int> no_children = {-1};    
+    const std::tuple<int,std::vector<int>> has_no_children = std::make_tuple(-1, no_children);
+    for (int cell = 0; cell < data_[0]-> size(0); ++cell){
+        l0_parent_to_children_cells[cell] = has_no_children;
```

Comment:
I think we should use the default directly during resize above (line 1404), i.e. with my suggestion above
```
l0_parent_to_children_cells.resize(data_[0]-> size(0), {{-1, {}});
```

Reply:
Done!

Reply:
Similar modification done for chid_to_parent_cells_ for the leafview:

` // For cells that do not have a parent, we set {-1,-1} by defualt and rewrite later for actual children
    leaf_child_to_parent_cells.resize(cell_count, std::array<int,2>({-1,-1}));`